### PR TITLE
layers: Remove redundant validation

### PR DESCRIPTION
**Pull request by jeremy-lunarg**
Remove redundant validation from ValidateOrIncrementBoundObjects.

Change-Id: I98f9de491d83788cf0b19d4c606d239e1a915e58

### CODE REVIEW

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -4356,7 +4335,7 @@ static bool validateCommandBufferState(layer_data *dev_data, GLOBAL_CB_NODE *cb_
 static bool validateResources(layer_data *dev_data, GLOBAL_CB_NODE *cb_node) {
     bool skip_call = false;
 
-    skip_call |= ValidateOrIncrementBoundObjects(dev_data, cb_node, false);
+    IncrementBoundObjects(dev_data, cb_node, false);
```
Comment (Tony-LunarG) [inline layers/core_validation.cpp]: If validation is happening elsewhere, there's no reason to call IncrementBoundObjects here, nor does it need a bool parameter to tell it whether to validate or increment, it should always just increment

**Comment by jeremy-lunarg** [comment]
Good call, Tony. I've made the change you suggested.

File: layers/core_validation.cpp (hunk 1 above)
Comment (jeremy-lunarg) [inline layers/core_validation.cpp]: Fixed.

**Comment by jeremy-lunarg** [comment]
Took Chris' PR instead.

**Pull request closed without merging**